core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java

```diff
@@ -319,6 +323,32 @@ public Response getWebUIOverview() {
       mMetaMaster.getJournalSpaceMonitor().map(monitor ->
           response.setJournalDiskWarnings(monitor.getJournalDiskWarnings()));
 
+      Gauge entriesSinceGauge = MetricsSystem.METRIC_REGISTRY.getGauges()
+              .get(MetricKey.MASTER_JOURNAL_ENTRIES_SINCE_CHECKPOINT.getName());
+      Gauge lastCkPtGauge = MetricsSystem.METRIC_REGISTRY.getGauges()
+          .get(MetricKey.MASTER_JOURNAL_LAST_CHECKPOINT_TIME.getName());
+
+      if (entriesSinceGauge != null && lastCkPtGauge != null) {
+        long entriesSinceCkpt = (Long) entriesSinceGauge.getValue();
+        long lastCkptTime = (Long) lastCkPtGauge.getValue();
+        long timeSinceCkpt = System.currentTimeMillis() - lastCkptTime;
+        boolean over72Hrs = timeSinceCkpt > ServerConfiguration.getMs(
+            PropertyKey.MASTER_WEB_JOURNAL_CHECKPOINT_WARNING_THRESHOLD_TIME);
+        boolean passedThreshold = entriesSinceCkpt > ServerConfiguration
+            .getLong(PropertyKey.MASTER_JOURNAL_CHECKPOINT_PERIOD_ENTRIES);
+        if (lastCkptTime > 0 && passedThreshold && over72Hrs) {
+          String time = ZonedDateTime
+              .ofInstant(Instant.ofEpochMilli(lastCkptTime), ZoneOffset.UTC)
+              .format(DateTimeFormatter.ISO_INSTANT);
+          response.setJournalCheckpointTimeWarning(String.format("Journal has not checkpointed in "
```

Comment:
Will this be always printed for a single-master cluster?
Do we have checkpoint capability from webUI? I believe we should at least direct users to an appropriate action.

Reply:
On a single-master cluster, a user can remove this message by forcing a checkpoint with `alluxio fsadmin journal checkpoint`. We can direct users to run this command to force a checkpoint?

Reply:
Right. But do we want to?

Reply:
I would replace "It is recommended to use the fsadmin tool to force a checkpoint." with something like "No action is required with multiple masters. If using a single master, use fsadmin to force a checkpoint. User activity will be blocked."

Reply:
I think that message is good. Just one comment: 

> No action is required with multiple masters. ...

My only concern is that..what's the point in showing the warning if we don't want the user to do anything about it?

Reply:
Good point. Can you hide the action if you detect multiple masters from the config?

Reply:
I changed the advice displayed based on the config: https://github.com/Alluxio/alluxio/pull/13690/commits/a68f75c86c5b9a71509a578c0af9ff525c2bc453#diff-78fd86b461c2be256edc9a1da53a384a8d7ac413537c95e25891b32f33b90d2eL342

Reply:
thx